internal/api/front/perses/src/app/project/dashboard/dashboard-details/dashboard-details.component.ts

```diff
@@ -25,29 +30,74 @@ import { DashboardModel } from '../dashboard.model';
   styleUrls: ['./dashboard-details.component.scss']
 })
 export class DashboardDetailsComponent implements OnInit {
+  private readonly paramName = 'dashboard';
```

Comment:
So here I think I would create another component called `dashboard-panels` that would be dedicated to the display of the panels and the sections.

Like what I did for the display of the variable. I actually created a dedicated component for that called `dashboard-variables`. Doing that will reduce the complexity of this component.

Then I would also create a component dedicated to the display of the ChartLine. Because when we will have to create another kind of panel like the [Gauge](https://swimlane.github.io/ngx-charts/#/ngx-charts/gauge) then it will become a bit messy.

Reply:
So technically in your html for the dashboard-details you would have something like that: 

```html
<app-page>
  <app-page-header>
    <h1 *ngIf="!isLoading">{{dashboard.metadata.name}}</h1>
  </app-page-header>
  <app-page-content>
    <mat-spinner *ngIf="isLoading"></mat-spinner>
    <app-page-sub-content *ngIf="!isLoading">
      <app-dashboard-variables [datasource]="dashboard.spec.datasource" [duration]="dashboard.spec.duration"
                               [variables]="dashboard.spec.variables"></app-dashboard-variables>
      <app-dashbord-sections [datasource]="dashboard.spec.datasource" [duration]="dashboard.spec.duration"
                               [sections]="dashboard.spec.sections"> </app-dashboard-sections>
    </app-page-sub-content>
  </app-page-content>
</app-page>
```

Reply:
You would certainly have the manage of the duration in the component `dashboard-details`. But that would be the only thing it would manage by its own. The rest will leave to dedicated component.

Reply:
Ok !